index.js

```diff
@@ -1,109 +1,64 @@
 'use strict';
 
-var Plugin           = require('./lib/plugin');
-var TemplatePlugin   = require('./lib/template-plugin');
-var JavascriptPlugin = require('./lib/javascript-plugin');
-var debug            = require('debug')('ember-cli:registry');
-
-function Registry(plugins, app) {
-  this.registry = {
-    js: [],
-    css: [],
-    'minify-css': [],
-    template: []
-  };
-
-  this.instantiatedPlugins = [];
-  this.availablePlugins = plugins;
-  this.app = app;
-  this.pluginTypes = {
-    'js': JavascriptPlugin,
-    'template': TemplatePlugin
-  };
-}
-
-module.exports = Registry;
-
-Registry.prototype.extensionsForType = function(type) {
-  var registered = this.registeredForType(type);
+const debug = require('debug')('ember-cli:registry');
 
-  var extensions =  registered.reduce(function(memo, plugin) {
-    return memo.concat(plugin.ext);
-  }, [type]).filter(Boolean);
-
-  extensions = require('ember-cli-lodash-subset').uniq(extensions);
-
-  debug('extensions for type %s: %s', type, extensions);
+class Registry {
+  constructor(app) {
```

Comment:
the `defaultRegistry()` function is still calling this constructor with `new Registry(app.dependencies(), app)`. was that intended?